# 01 - data acquisition (sample)

load sample data from `data/samples/`, parse filenames, verify structure.

In [1]:
from pathlib import Path
import pandas as pd

import sys
sys.path.append('../../')
from src.config.settings import SAMPLES
from src.utils.helpers import parse_filename, assign_split

In [2]:
df = pd.read_csv(SAMPLES / "sample_labels.csv")
print(f"loaded {len(df)} sample files")

loaded 416 sample files


In [3]:
print(df.head())
print(df.dtypes)

                                            filepath channel  emotion_code  \
0  /mnt/d/career/projects/lightweight-speech-emot...    song             1   
1  /mnt/d/career/projects/lightweight-speech-emot...    song             1   
2  /mnt/d/career/projects/lightweight-speech-emot...    song             1   
3  /mnt/d/career/projects/lightweight-speech-emot...    song             1   
4  /mnt/d/career/projects/lightweight-speech-emot...    song             2   

   emotion intensity  statement  repetition  actor gender split  
0  neutral    normal          1           1     23   male  test  
1  neutral    normal          1           2     23   male  test  
2  neutral    normal          2           1     23   male  test  
3  neutral    normal          2           2     23   male  test  
4     calm    normal          1           1     23   male  test  
filepath          str
channel           str
emotion_code    int64
emotion           str
intensity         str
statement       int64
rep

In [4]:
print(df.groupby(['split','channel']).size().unstack(fill_value=0))

channel  song  speech
split                
test       44      60
train      88     120
val        44      60


In [5]:
print("actors per split:")
for s in ['train','val','test']:
    a = sorted(df[df['split']==s]['actor'].unique())
    print(f"  {s}: {a} ({len(a)} actors)")

actors per split:
  train: [np.int64(1), np.int64(2)] (2 actors)
  val: [np.int64(20)] (1 actors)
  test: [np.int64(23)] (1 actors)


In [6]:
print("file exists check:")
all_ok = all(Path(p).exists() for p in df['filepath'])
print(f"  all {len(df)} files exist: {all_ok}")

file exists check:
  all 416 files exist: True


sample data ready. 416 files across 4 actors, all splits represented.